doc/xds-test-descriptions.md

```diff
@@ -621,3 +621,110 @@ There are four sub-tests:
       `rpc-behavior: sleep-4`.
    1. Test driver asserts client recieves ~100% status `OK` for EmptyCall
       and ~100% status `DEADLINE_EXCEEDED` for UnaryCall.
+
+### api_listener
+The test case verifies a specific use case where it creates a second TD API 
+listener using the same name as the existing one and then delete the old one. 
+The test driver verifies this is a safe way to update the API listener 
+configuration while keep using the existing name.
+
+Client parameters:
+
+1.  --num_channels=1
+1.  --qps=100
+
+Load balancer configuration:
+
+1.  One MIG with two backends.
+
+Assert:
+
+The test driver configuration steps:
+1. The test driver creates the first set of forwarding rule + target proxy + 
+URL map with a test host name.
+1. Then the test driver creates a second set of forwarding rule + target proxy + 
+URL map with the same test host name.
+1. The test driver deletes the first set of configurations in step 1.
+
+The test driver verifies, at each configuration step, the traffic is always able 
+to reach the designated hosts.
+
+### metadata_filter
+This test case verifies that metadata filter configurations in URL map match 
+rule are effective at Traffic Director for routing selection against downstream
+node metadata.
+
+Client parameters:
+
+1.  --num_channels=1
+1.  --qps=100
+
+Load balancer configuration:
+
+1.  Two MIGs in the same zone, each having two backends.
+
+There are four test sub-cases:
+1. Test `MATCH_ALL` metadata filter criteria.
+1. Test `MATCH_ANY` metadata filter criteria.
+1. Test mixed `MATCH_ALL` and `MATCH_ANY` metadata filter criteria.
+1. Test when multiple match rules with metadata filter all match.
+
+Assert:
+
+At each test sub-case described above, the test driver configures
+and verifies:
+
+1. Set default URL map, and verify traffic goes to the original backend hosts. 
+1. Then patch URL map to update the match rule with metadata filter 
```

Comment:
It seems this should say "_add_ a match rule _for the alternate MIG_". And (1) should say "Set a default URL map pointing to one of the MIGs"

It isn't at all clear what configuration is expected here. Is this only positive testing? As in, let's say that the server ignored the metadata match criteria (essentially becoming wildcard), wouldn't this test still pass? What filter labels are to be used? Only a single label, or multiple labels? Is it okay if all labels match for the MATCH_ANY?